spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/cassandra/CassandraDataAutoConfiguration.java

```diff
@@ -80,10 +113,41 @@ public CassandraSessionFactoryBean session(CassandraConverter converter)
 		session.setCluster(this.cluster);
 		session.setConverter(converter);
 		session.setKeyspaceName(this.properties.getKeyspaceName());
-		session.setSchemaAction(SchemaAction.NONE);
+		session.setSchemaAction(SchemaAction.valueOf(this.properties.getSchemaAction()));
```

Comment:
No, the idea is to store the String value like you did, but ask the binding to retrieve `SchemaAction` in the auto-configuration, once we know we have access to the class. It has been kind of a blocker so far actually. I guess there is a way to do that without it but it might involve copy/pasting some code.